docs/naming_conventions.md

```diff
@@ -0,0 +1,151 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+1. The `id` attribute for a resource in the FHIR model should be
+   populated with some kind of logical value that describes the semantics of
+   the resource (e.g. ResearchParticipant vs 2349703).
+
+   **NOTE:** This ONLY applies to the files in the FHIR model. Non-conformance
+   resources (representing real data) loaded into the FHIR server will likely
+   have an autogenerated alphanumeric ID which may or may not be random.
+
+2. The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
+
+3. All resource `id` values should have the same casing. There does not seem
+   to be a consistent standard or recommendation for this but we should pick
+   one and stick to it.
+
+   Based on observations, the FHIR spec seems to use PascalCase for
+   resource types (e.g. `Patient`, `SearchParameter`) and camelCase
+   (e.g. `gender`, `birthDate`) for resource attributes
+
+   Other FHIR models have used one of or a mix of the following:
+
+   - kebab-case (e.g. `age-at-event`, `sequencing-experiment`)
+   - PascalCase (see above)
+   - camelCase (see above)
+
+## Resource File Name
+
+### Conformance Resources
+
+All conformance resources should follow this file naming pattern:
+
+**Pattern**
+```
+<resource type>_<base definition resource id>_<resource id>.json
```

Comment:
Arguments against including the base definition ID:

1. You won't have more than one base definition ID per resource ID because the resource ID needs to be unique, so the resource ID already implies the base ID.
2. Including it means that you don't know what to do below for non-conformance resources with multiple meta profiles.